File: npm/src/commands/log.js
'use strict'
const { Command } = require('commander')
const fs = require('fs')
const path = require('path')

const cmd = new Command('log')
cmd.description('Show roadmap state transition history')
cmd.option('--tail <n>', 'Number of recent transitions to show', '20')
cmd.action(async (opts) => {
  const tail = parseInt(opts.tail, 10)
  const logPath = path.join('docs', 'roadmaps', '.trackfw-log')

  if (!fs.existsSync(logPath)) {
    console.log('No transitions recorded yet.')
    return
  }

  const lines = fs.readFileSync(logPath, 'utf8')
    .split('\n')
    .filter(l => l.trim() !== '')

  const start = Math.max(0, lines.length - tail)
  const visible = lines.slice(start)

  console.log('── trackfw log ─────────────────────────')
  visible.forEach(l => console.log(l))
})

module.exports = cmd

File: internal/commands/log.go
package commands

import (
	"bufio"
	"fmt"
	"os"

	"github.com/spf13/cobra"
)

func newLogCmd() *cobra.Command {
	var tail int
	cmd := &cobra.Command{
		Use:   "log",
		Short: "Show roadmap state transition history",
		RunE: func(cmd *cobra.Command, args []string) error {
			return runLog(tail)
		},
	}
	cmd.Flags().IntVar(&tail, "tail", 20, "Number of recent transitions to show")
	return cmd
}

func runLog(tail int) error {
	f, err := os.Open("docs/roadmaps/.trackfw-log")
	if os.IsNotExist(err) {
		fmt.Println("No transitions recorded yet.")
		return nil
	}
	if err != nil {
		return err
	}
	defer f.Close()

	var lines []string
	scanner := bufio.NewScanner(f)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	start := 0
	if len(lines) > tail {
		start = len(lines) - tail
	}

	fmt.Println("── trackfw log ─────────────────────────")
	for _, line := range lines[start:] {
		fmt.Println(line)
	}
	return nil
}
